lib/src/puppetdb-cli.cc

```diff
@@ -107,7 +121,85 @@ PuppetDBConn::parseServerUrls(const json::JsonContainer& config) {
     }
 }
 
-size_t write_data(void *ptr, size_t size, size_t nmemb, FILE *stream) {
+class JsonQueue {
```

Comment:
This whole class uses *super* fine-grained locking as well, which cannot possibly be fast.